core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -598,26 +598,24 @@ public void onResponse(Frame responseFrame) {
         if (responseMessage instanceof SchemaChange) {
           SchemaChange schemaChange = (SchemaChange) responseMessage;
           context
-              .getTopologyMonitor()
-              .checkSchemaAgreement()
-              .thenCombine(
-                  context
-                      .getMetadataManager()
-                      .refreshSchema(schemaChange.keyspace, false, false)
```

Comment:
Also, `thenCombine` was wrong here, because it started the refresh before the agreement was done.

Reply:
I was about to ask if there are implications in refreshing the schema before agreement is reached, but it seems we've been in that boat for a while now.

Reply:
I guess it would manifest as the control node not having the new table yet, we've never observed it in practice.